visualization/app/codeCharta/ui/codeMap/codeMap.preRender.service.ts

```diff
@@ -58,52 +60,68 @@ export class CodeMapPreRenderService implements SettingsServiceSubscriber, FileS
 	}
 
 	public getRenderMap(): CodeMapNode {
-		return this.lastRender.renderFile ? this.lastRender.renderFile.map : null
+		return this.lastRender.map
 	}
 
 	public getRenderFileMeta(): FileMeta {
-		return this.lastRender.renderFile.fileMeta
+		return this.lastRender.fileMeta
 	}
 
 	public onSettingsChanged(settings: Settings, update: RecursivePartial<Settings>, event: angular.IAngularEvent) {
 		this.lastRender.settings = settings
 
 		if (this.lastRender.fileStates && update.fileSettings && (update.fileSettings.blacklist || update.fileSettings.markedPackages)) {
-			this.lastRender.renderFile = this.getSelectedFilesAsUnifiedMap(this.lastRender.fileStates)
-			this.lastRender.renderFile.settings.fileSettings = settings.fileSettings
+			this.updateRenderMapAndFileMeta()
 			this.decorateIfPossible()
 		}
-		this.renderIfRenderObjectIsComplete()
+
+		if (this.allNecessaryRenderDataAvailable()) {
+			if (this.settingsOnlyContainNewScaling(update)) {
+				this.callScaleMap()
```

Comment:
why `CALLScaleMap`? I'd just call it `scaleMap`

Reply:
I renamed it to `scaleMapAndNotify()`